src/platform/qt/library/LibraryController.cpp

```diff
@@ -213,16 +248,46 @@ void LibraryController::loadDirectory(const QString& dir, bool recursive) {
 	mLibraryLoadDirectory(library.get(), dir.toUtf8().constData(), recursive);
 	m_libraryJob.testAndSetOrdered(libraryJob, -1);
 }
+
 void LibraryController::setShowFilename(bool showFilename) {
 	if (showFilename == m_showFilename) {
 		return;
 	}
 	m_showFilename = showFilename;
-	if (m_libraryGrid) {
-		m_libraryGrid->setShowFilename(m_showFilename);
+	m_libraryModel->setShowFilename(m_showFilename);
+	refresh();
+}
+
+void LibraryController::resizeEvent(QResizeEvent*) {
+	resizeTreeView(false);
+}
+
+void LibraryController::resizeTreeView(bool expand) {
+	if (expand) {
+		m_treeView->expandAll();
 	}
-	if (m_libraryTree) {
-		m_libraryTree->setShowFilename(m_showFilename);
+
+	int viewportWidth = m_treeView->viewport()->width();
```

Comment:
Not sure if it's still a problem, but I had issues with the column sizes at certain window sizes--I think if the window was too narrow. Worth checking different window sizes.

Reply:
That should work? This puts a minimum usable size on the column so it'll just scroll horizontally if necessary.

Reply:
I remembered what the bug was: if you start with the library not displayed, then enable it in settings, the resizing doesn't trigger. I had a small window so I thought that was related.

Reply:
Oh, interesting. Easy enough to fix.